### PR TITLE
Implemented basic memory operations

### PR DESCRIPTION
**Pull request by kryczkal**
Memory manipulation functions:
* [`alkos/libc/include/memory.h`](diffhunk://#diff-9f68c1d9dfef3e299211e119c6c54390213a9afbfa6575775977d64d6fc56570R1-R56): Added declarations for memory manipulation functions: `memcpy`, `memmove`, `memset`, and `memcmp`.
* [`alkos/libc/memory/memory.cpp`](diffhunk://#diff-3cc2c6ce0a81e49d5021e0d14baf0ed8230f1c7964312d2b3e49e140dfbb363cR1-R76): Implemented the memory manipulation functions `memcpy`, `memmove`, `memset`, and `memcmp`.

### CODE REVIEW

File: alkos/libc/include/memory.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * @brief Copy memory area as if it was first copied to a temporary area and then copied back to dest
+ *
+ * @param dest Destination memory area
+ * @param src Source memory area
+ * @param n Number of bytes to copy
+ * @return void* Pointer to the destination memory area
+ */
+void *memmove(void *dest, const void *src, size_t n);
+/**
```
Comment (F1r3D3v) [inline alkos/libc/include/memory.h:31]: Missing newline

File: alkos/libc/memory/memory.cpp
Comment (F1r3D3v) [inline alkos/libc/memory/memory.cpp:1]: I suggest maintaining consistent spacing between functions, such as adding a newline between loop and variable declarations.

**Review by F1r3D3v — APPROVED** [review]
Overall, great job! Just a few small changes suggested.

File: alkos/libc/memory/memory.cpp
Comment (kryczkal) [reply alkos/libc/memory/memory.cpp:1]: done

**Pull request merged**